In [9]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.linear_model import LinearRegression
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_squared_error
import numpy as np

In [10]:
# Read the CSV file into a DataFrame
df = pd.read_csv('labelled_comments.csv')  # Replace 'your_dataset.csv' with your actual file path

# Split the data into training and testing sets
train_data, test_data, train_labels, test_labels = train_test_split(df['comment'], df['sentiment'], test_size=0.2, random_state=42)

# Convert comments to TF-IDF features
vectorizer = TfidfVectorizer()
train_features = vectorizer.fit_transform(train_data)
test_features = vectorizer.transform(test_data)

# Train a Random Forest Regressor model
model = RandomForestRegressor(n_estimators=100, random_state=42)
model.fit(train_features, train_labels)

# Predict on the test set
predictions = model.predict(test_features)

# Evaluate the model
mse = mean_squared_error(test_labels, predictions)
rmse = np.sqrt(mse)
print(f"Mean Squared Error: {mse}")
print(f"Root Mean Squared Error: {rmse}")

Mean Squared Error: 0.048405059045893824
Root Mean Squared Error: 0.22001149753113772


## For LinearRegression:
- Mean Squared Error: 0.1323086672735092
- Root Mean Squared Error: 0.363742583805511

## For RandomForestRegression
- Mean Squared Error: 0.048405059045893824
- Root Mean Squared Error: 0.22001149753113772